### ISSUE TITLE
Thread Performance warning in Xcode14

### ISSUE THREAD

**Issue by duyenlv** [aws-amplify/aws-sdk-ios#4360]
**Describe the bug**

#### Hi AWS team, can you check and fix following issue:
This is screenshot and logs I encountered when upgrading my iOS project to Xcode14

![Screen Shot 2022-10-24 at 10 32 24](https://user-images.githubusercontent.com/2205362/197443706-6627a2be-bb5b-4f1e-896e-f699516d5f80.png)

```
Thread Performance Checker: Thread running at QOS_CLASS_USER_INTERACTIVE waiting on a lower QoS thread running at QOS_CLASS_DEFAULT. Investigate ways to avoid priority inversions
PID: 968, TID: 32084
Backtrace
=================================================================
3   AWSIoT                              0x00000001047841c4 -[_SRRunLoopThread runLoop] + 44
4   AWSIoT                              0x0000000104783f10 __46+[NSRunLoop(SRWebSocket) AWSSR_networkRunLoop]_block_invoke + 124
5   libdispatch.dylib                   0x0000000104dd98c4 _dispatch_client_callout + 16
6   libdispatch.dylib                   0x0000000104ddb208 _dispatch_once_callout + 80
7   AWSIoT                              0x0000000104783e6c +[NSRunLoop(SRWebSocket) AWSSR_networkRunLoop] + 84
8   AWSIoT                              0x000000010477d648 -[AWSSRWebSocket openConnection] + 80
9   AWSIoT                              0x000000010477ba08 -[AWSSRWebSocket open] + 216
10  AWSIoT                              0x00000001047452e8 -[AWSIoTMQTTClient initWebSocketConnectionForURL:] + 1580
11  AWSIoT                              0x0000000104744c68 __48-[AWSIoTMQTTClient webSocketConnectWithClientId]_block_invoke.123 + 876
12  AWSCore                             0x0000000103efef6c __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke + 88
13  AWSCore                             0x0000000103eb6990 __30+[AWSExecutor defaultExecutor]_block_invoke_2 + 148
14  AWSCore                             0x0000000103eb7160 -[AWSExecutor execute:] + 88
15  AWSCore                             0x0000000103eff2c0 __56-[AWSTask continueWithExecutor:block:cancellationToken:]_block_invoke.15 + 44
16  AWSCore                             0x0000000103efea34 -[AWSTask runContinuations] + 452
17  AWSCore                             0x0000000103efe444 -[AWSTask trySetResult:] + 208
18  AWSCore                             0x0000000103effff0 -[AWSTaskCompletionSource setResult:] + 76
19  Treau                               0x0000000102dd55b8 $s5Treau29IoTSessionCredentialsProviderC11credentialsSo7AWSTaskCySo14AWSCredentialsCGyFyAA9APIOutputOyAA19CognitoSessionModelVAA15DefaultAPIErrorOGcfU_yyScMYccfU_ + 340
20  Treau                               0x0000000102cfcefc $sIeg_IeyB_TR + 48
21  libdispatch.dylib                   0x0000000104dd808c _dispatch_call_block_and_release + 24
22  libdispatch.dylib                   0x0000000104dd98c4 _dispatch_client_callout + 16
23  libdispatch.dylib                   0x0000000104de8428 _dispatch_main_queue_drain + 920
24  libdispatch.dylib                   0x0000000104de8080 _dispatch_main_queue_callback_4CF + 40
25  CoreFoundation                      0x00000001cce40ac8 5F4831C9-4949-34EA-B373-3F45C312B753 + 604872
26  CoreFoundation                      0x00000001cce24754 5F4831C9-4949-34EA-B373-3F45C312B753 + 489300
27  CoreFoundation                      0x00000001cce28dfc CFRunLoopRunSpecific + 584
28  GraphicsServices                    0x0000000206cda984 GSEventRunModal + 160
29  UIKitCore                           0x00000001d4008e7c 0A5F2FF1-A67F-3263-BAFC-1224B1501A17 + 3632764
30  UIKitCore                           0x00000001d4008af4 UIApplicationMain + 312
31  Treau                               0x0000000102da1ce4 main + 64
32  dyld                                0x00000001ee4e2de0 761F47F5-A570-31C3-AC49-AF3DED335020 + 81376
```

**Environment(please complete the following information):**
 - SDK Version: 2.28.0
 - Dependency Manager: Cocoapods
 - Swift Version : 5
 - Xcode Version: 14.0.1 (Apple Silicon)

**Device Information (please complete the following information):**
 - iOS Version: 16

**Comment by l1Dan** [comment]
the same problem

<img width="1893" alt="image" src="https://user-images.githubusercontent.com/7961100/201614021-0df8a70e-8498-4c9d-aa9c-d9d64fdb25a4.png">

**Comment by duyenlv** [comment]
Anyone help?

**Comment by atierian** [comment]
This has been addressed in https://github.com/aws-amplify/aws-sdk-ios/pull/4575 and will be included in the next release. We'll update this issue once that release has been completed. Thanks for reporting this!

**Comment by atierian** [comment]
The fix for this has been released in [2.30.3](https://github.com/aws-amplify/aws-sdk-ios/releases/tag/2.30.3)
I'm closing this issue. If you still experience the runtime thread performance warning after upgrading, please comment here and we'll reopen to investigate further.

**Comment by mzying2013** [comment]
AWSIoT (2.35.0)
Xcode Version 15.3 (15E204a)
This problem has reappeared, please help me.

Thread Performance Checker: Thread running at User-interactive quality-of-service class waiting on a lower QoS thread running at Default quality-of-service class. Investigate ways to avoid priority inversions
PID: 6817, TID: 2999505
Backtrace
=================================================================
3   DY                              0x0000000102cdaa34 -[AWSDDLog queueLogMessage:asynchronously:] + 76
4   DY                              0x0000000102cdb320 -[AWSDDLog log:message:level:flag:context:file:function:line:tag:] + 460
5   DY                              0x0000000102cdb128 +[AWSDDLog log:message:level:flag:context:file:function:line:tag:] + 260
6   DY                              0x0000000102cdac6c +[AWSDDLog log:level:flag:context:file:function:line:tag:format:] + 260
7   DY                              0x0000000102e39988 -[AWSMQTTSession timerHandler:] + 1748
8   Foundation                          0x00000001a6a76ce0 __NSFireTimer + 96
9   CoreFoundation                      0x00000001a753b6e4 3A5F992A-D1CD-312E-BD2E-F7C66343A417 + 710372
10  CoreFoundation                      0x00000001a753b388 3A5F992A-D1CD-312E-BD2E-F7C66343A417 + 709512
11  CoreFoundation                      0x00000001a74c5184 3A5F992A-D1CD-312E-BD2E-F7C66343A417 + 225668
12  CoreFoundation                      0x00000001a74c218c 3A5F992A-D1CD-312E-BD2E-F7C66343A417 + 213388
13  CoreFoundation                      0x00000001a74c1968 CFRunLoopRunSpecific + 608
14  Foundation                          0x00000001a63504a8 D92E19C1-6299-3E94-8614-C505D5ABCCDB + 177320
15  DY                              0x0000000102e3335c -[AWSIoTStreamThread main] + 956
16  Foundation                          0x00000001a63c1a9c D92E19C1-6299-3E94-8614-C505D5ABCCDB + 641692
17  libsystem_pthread.dylib             0x0000000203471a90 _pthread_start + 136
18  libsystem_pthread.dylib             0x0000000203470fcc thread_start + 8

**Comment by lawmicha** [comment]
Hi @mzying2013, can you open a new issue using the [issue template](https://github.com/aws-amplify/aws-sdk-ios/issues/new?assignees=&labels=&projects=&template=bug_report.md&title=) ? This will help us gather any additional details needed and a new issue will help us track this better. Feel free to reference this issue in the new one
